wp-includes/class-wp-service-workers.php

```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+ * Dependencies API: WP_Service_Workers class
+ *
+ * @since ?
+ *
+ * @package PWA
+ */
+
+/**
+ * Class used to register service workers.
+ *
+ * @since ?
+ *
+ * @see WP_Dependencies
+ */
+class WP_Service_Workers extends WP_Scripts {
+
+	/**
+	 * Param for service workers.
+	 *
+	 * @var string
+	 */
+	public $query_var = 'wp_service_worker';
+
+	/**
+	 * Output for service worker scope script.
+	 *
+	 * @var string
+	 */
+	public $output = '';
+
+	/**
+	 * Initialize the class.
+	 */
+	public function init() {
+		/**
+		 * Fires when the WP_Service_Workers instance is initialized.
+		 *
+		 * @param WP_Service_Workers $this WP_Service_Workers instance (passed by reference).
+		 */
+		do_action_ref_array( 'wp_default_service_workers', array( &$this ) );
+	}
+
+	/**
+	 * Register service worker.
+	 *
+	 * Registers service worker if no item of that name already exists.
+	 *
+	 * @param string          $handle Name of the item. Should be unique.
+	 * @param string|callable $src    URL to the source in the WordPress install, or a callback that returns the JS to include in the service worker.
+	 * @param array           $deps   Optional. An array of registered item handles this item depends on. Default empty array.
+	 * @param array           $scopes Optional. Scopes of the service worker. Default relative path.
+	 * @return bool Whether the item has been registered. True on success, false on failure.
+	 */
+	public function register( $handle, $src, $deps = array(), $scopes = array() ) {
+
+		// Set default scope if missing.
+		if ( empty( $scopes ) ) {
+			$scopes = array( site_url( '/', 'relative' ) );
+		}
+		return parent::add( $handle, $src, $deps, false, compact( 'scopes' ) );
+	}
+
+	/**
+	 * Get service worker logic for scope.
+	 *
+	 * @param string $scope Scope of the Service Worker.
+	 */
+	public function serve_request( $scope ) {
+
+		header( 'Content-Type: text/javascript; charset=utf-8' );
+		nocache_headers();
+
+		if ( ! in_array( $scope, $this->get_scopes(), true ) ) {
+			status_header( 404 );
```

Comment:
See #26. This should perhaps not return a `404` but rather return a `200` and a service worker script that simply unregisters itself.

Reply:
(My suggestion won't work.)